tests/test_main.py

```diff
@@ -190,3 +190,64 @@ def test_write_results_from_dummy_data_file(dummy_data_fname, results_fname, tmp
     results_file = tmp_path / results_fname
     main.write_results(dummy_data_file, results_file)
     assert results_file.read_text(encoding="utf-8") == "id\n1\n2\n"
+
+
+@pytest.mark.parametrize(
+    "sql_query",
+    [
+        # A complex query with a common table expression and a correlated
+        # subquery and column aliases to ensure that the parser
+        # correctly identifies the output columns
+        """
+        ;with CTE as (
+        SELECT
+            column1,
+            [column 2],
+            count(*) as column_3
+        FROM
+            table
+        GROUP BY
+            column1,
+            [column 2],
+        )
+        SELECT c.*,x.column_4
+        FROM CTE c
+        JOIN (
+            SELECT
+                column1,
+                column_4
+            FROM table2
+        ) x
+            ON c.column1 = x.column1
+        """,
+        # A multi-statement batch to ensure we create headers for the last
+        # statement in the batch
+        """
+        SELECT foo, bar, baz, qux
+        INTO #temptable;
+        SELECT
+            foo as [column1],
+            bar as [column 2],
+            baz as [column_3],
+            qux as [column_4]
+        FROM #temptable
+        """,
+    ],
+)
+def test_generate_column_headers(sql_query):
+    results = main.generate_column_headers(sql_query)
+    header = next(results)
+    assert list(header.keys()) == ["column1", "column 2", "column_3", "column_4"]
+    # we want only a header, no row data
+    assert not next(results, None)
```

Comment:
This is succinct, but I don't think it's as easy to reason about as...

```suggestion
    results = list(main.generate_column_headers(sql_query))
    assert len(results) == 1
    assert results[0].keys() == ["column1", "column 2", "column_3", "column_4"]
```

However, I don't think `generate_column_headers` should yield a dict.